src/net_processing.cpp

```diff
@@ -3305,6 +3314,43 @@ bool PeerLogicValidation::SendMessages(CNode* pto, std::atomic<bool>& interruptM
     return true;
 }
 
+void CheckLocalServices(ServiceFlags& nLocalServices) {
+    if (fPruneMode) {
+        LOCK(cs_main);
+
+        // make sure NODE_NETWORK is disabled
+        nLocalServices = ServiceFlags(nLocalServices & ~NODE_NETWORK);
+
+        // don't signal NODE_NETWORK_LIMITED during initial block download
+        if (IsInitialBlockDownload()) {
+            return;
+        }
+
+        // check how many blocks we have available (backwards from tip)
+        // don't go futher down then chainActive.Height()-NODE_NETWORK_LIMITED
```

Comment:
This seems entirely redundant - even if by super rare chance we end up due to reorg short of 288 blocks, we won't be *very* short of 288 blocks, and will be caught back up within the next block or two, might as well just always set NODE_NETWORK_LIMITED (you can static_assert MIN_BLOCKS_TO_KEEP is at least 288 if you really want to have a check).